### PR TITLE
lib: lte_link_control: Improve Kconfig option naming

### PR DESCRIPTION
**Pull request by tokangas**
Renamed `CONFIG_LTE_MODE_PREFERENCE` to `CONFIG_LTE_MODE_PREFERENCE_VALUE` to allow the LTE mode preference choice to be named consistently.

Added a name (`CONFIG_LTE_MODE_PREFERENCE`) for the LTE mode preference choice.

Renamed `CONFIG_LTE_NETWORK_DEFAULT` to `CONFIG_LTE_NETWORK_MODE_DEFAULT` to make the naming consistent.

### CODE REVIEW

File: lib/lte_link_control/Kconfig [hunk 1]
```diff
@@ -228,7 +228,7 @@ config LTE_MODE_PREFERENCE_NBIOT_PLMN_PRIO
 
 endchoice
 
-config LTE_MODE_PREFERENCE
```
Comment (trantanen) [inline lib/lte_link_control/Kconfig:231]: Please mention in the commit message and PR description that this has been renamed and taken into use for another purpose.
And changelog entry for the same story.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -679,6 +679,9 @@ Modem libraries
       The application uses the Kconfig option :kconfig:option:`CONFIG_LTE_RAI_REQ` and ``SO_RAI_*`` socket options instead.
     * The CE level enum names for :c:enum:`lte_lc_ce_level` to not include the number of repetitions.
     * The default network mode from :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M` to :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_GPS`.
+    * The ``CONFIG_LTE_MODE_PREFERENCE`` Kconfig option has been renamed to :kconfig:option:`CONFIG_LTE_MODE_PREFERENCE_VALUE`.
+    * The ``CONFIG_LTE_NETWORK_DEFAULT`` Kconfig option has been renamed to :kconfig:option:`CONFIG_LTE_NETWORK_MODE_DEFAULT`.
+    * The LTE mode preference Kconfig choice has been added a name, :kconfig:option:`CONFIG_LTE_MODE_PREFERENCE`.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:684]:
```suggestion
    * The LTE mode preference Kconfig choice has been named as ``CONFIG_LTE_MODE_PREFERENCE``.
```
This is a bit confusing. You just renamed this Kconfig option but then you use it as a choice name. Not sure if these choice names can be tagged with :kconfig:option:.

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (tokangas) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:684]: Yes, renaming was done to be able to name the choice logically. `CONFIG_LTE_MODE_PREFERENCE_VALUE` is only used by LTE LC internally and modem_shell, it's probably not used widely anyway. And I'm not sure either if the choice name can be referenced, we'll see.

Comment (tokangas) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:684]: At least the doc build was successful.

**Review by jhirsi — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Comment by tokangas** [comment]
There's currently a problem with `test-ci-nrfconnect-boot-fw-update`, causing all runs to fail. Waiting for the issue to be resolved before re-running CI.

**Pull request merged**